test/new-e2e/tests/agent-shared-components/api/api_test.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	agentCmdPort = 5001
+)
+
+type apiSuite struct {
+	e2e.BaseSuite[environments.Host]
+}
+
+func TestApiSuite(t *testing.T) {
+	e2e.Run(t, &apiSuite{}, e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake()))
+}
+
+type agentEndpointInfo struct {
+	name     string
+	scheme   string
+	port     int
+	endpoint string
+	method   string
+	data     string
+}
+
+func (endpointInfo *agentEndpointInfo) url() *url.URL {
+	return &url.URL{
+		Scheme: endpointInfo.scheme,
+		Host:   net.JoinHostPort("localhost", strconv.Itoa(endpointInfo.port)),
+		Path:   endpointInfo.endpoint,
+	}
+}
+
+func (endpointInfo *agentEndpointInfo) fetchCommand(authtoken string) string {
+	var data string
+	if len(endpointInfo.data) > 0 {
+		data = endpointInfo.data
+	} else {
+		data = "{}"
+	}
+
+	// -L: follow redirects
+	// -s: silent
+	// -k: allow insecure server connections
+	// -H: add a header
+	// -d: request data (json)
+	return fmt.Sprintf(
+		`curl -s -k -H "authorization: Bearer %s" -X %s "%s" -d "%s"`,
+		authtoken,
+		endpointInfo.method,
+		endpointInfo.url().String(),
+		data,
+	)
+}
+
+func (v *apiSuite) TestInternalAgentAPIEndpoints() {
+	testcases := []struct {
+		endpoint agentEndpointInfo
```

Comment:
```suggestion
	agentEndpointInfo
```

If you embed the `agentEndpointInfo` instead, you can avoid a level of nesting for each of the test cases.

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/24616/commits/f40c2fbe7dc338ccdb382c1a2fbc727e1842d84b